t_modify_http_headers/test_logic.py

```diff
@@ -0,0 +1,355 @@
+                else generate_http1_request(optional_headers),
+                "200",
+            )
+
+        expected_response = self.check_response(optional_headers, expected_headers, client)
+        expected_request = self.get_expected_request(optional_headers, expected_headers, client)
+
+        self.assertEqual(expected_response, client.last_response)
+        self.assertEqual(expected_request, server.last_request)
+
+        return client, server
+
+    def check_response(
+        self, optional_headers: list, expected_headers: list, client
+    ) -> Response or H2Response:
+        if client.proto == "h2":
+            tempesta_headers = [
+                ("via", "2.0 tempesta_fw (Tempesta FW pre-0.7.0)"),
+            ]
+        else:
+            tempesta_headers = [
+                ("via", "1.1 tempesta_fw (Tempesta FW pre-0.7.0)"),
+                ("Connection", "keep-alive"),
+            ]
+
+        if self.cache:
+            tempesta_headers.append(("age", client.last_response.headers["age"]))
+
+        if self.directive == "req":
+            expected_response = generate_response(
+                optional_headers=tempesta_headers + optional_headers
+            )
+        else:
+            expected_response = generate_response(
+                optional_headers=tempesta_headers + expected_headers
+            )
+
+        if client.proto == "h2":
+            expected_response = H2Response(
+                expected_response.replace("HTTP/1.1 200 OK", ":status: 200")
+            )
+        else:
+            expected_response = Response(expected_response)
+        expected_response.set_expected()
+        return expected_response
+
+    def get_expected_request(
+        self, optional_headers: list, expected_headers: list, client
+    ) -> Request:
+        tempesta_headers = [
+            ("X-Forwarded-For", "127.0.0.1"),
+            ("via", "1.1 tempesta_fw (Tempesta FW pre-0.7.0)"),
+        ]
+        if self.directive == "req":
+            expected_request = generate_http1_request(
+                optional_headers=tempesta_headers + expected_headers
+            )
+        else:
+            expected_request = generate_http1_request(
+                optional_headers=tempesta_headers + optional_headers
+            )
+
+        if client.proto == "h2":
+            expected_request = expected_request.replace("Connection: keep-alive\r\n", "")
+
+        expected_request = Request(expected_request)
+        expected_request.set_expected()
+
+        return expected_request
+
+    def test_set_non_exist_header(self):
+        """
+        Header will be added in request/response if it is missing from base request/response.
+        """
+        self.base_scenario(
+            config=f'{self.directive}_hdr_set non-exist-header "qwe";\n',
+            optional_headers=[],
+            expected_headers=[("non-exist-header", "qwe")],
+        )
+
+    def test_add_non_exist_header(self):
```

Comment:
What a difference between `test_add_non_exist_header` and `test_add_one_hdr` from `test_common_config.py`? Looks like duplicated tests.

Reply:
It's duplicate. Removed.